### ISSUE TITLE
Test failure

### ISSUE THREAD

**Issue by fabaff** [sissaschool/xmlschema#272]
During the package update for NixOS we see a failure of the test mentioned below.

```bash
=================================== FAILURES ===================================
__________________ TestXsd11SimpleTypes.test_assertion_facet ___________________

self = <tests.validators.test_simple_types.TestXsd11SimpleTypes testMethod=test_assertion_facet>

    def test_assertion_facet(self):
        self.check_schema("""
            <xs:simpleType name='DimensionType'>
              <xs:restriction base='xs:integer'>
                <xs:assertion test='string-length($value) &lt; 2'/>
              </xs:restriction>
            </xs:simpleType>""")
    
        schema = self.check_schema("""
            <xs:simpleType name='MeasureType'>
              <xs:restriction base='xs:integer'>
                <xs:assertion test='$value &gt; 0'/>
              </xs:restriction>
            </xs:simpleType>""")
        self.assertTrue(schema.types['MeasureType'].is_valid('10'))
        self.assertFalse(schema.types['MeasureType'].is_valid('-1.5'))
    
        # Schema is valid but data value can't be compared with the string on the right
        schema = self.check_schema("""
            <xs:simpleType name='RestrictedDateTimeType'>
              <xs:restriction base='xs:dateTime'>
                <xs:assertion test="$value > '1999-12-31T23:59:59'"/>
              </xs:restriction>
            </xs:simpleType>""")
        self.assertFalse(schema.types['RestrictedDateTimeType'].is_valid('2000-01-01T12:00:00'))
    
        with self.assertRaises(XMLSchemaValidationError) as ctx:
            schema.types['RestrictedDateTimeType'].validate('2000-01-01T12:00:00')
>       self.assertIn("wrong type <class 'str'> for operand", str(ctx.exception))
E       AssertionError: "wrong type <class 'str'> for operand" not found in 'failed validating DateTime(2000, 1, 1, 12, 0, 0) with XsdAssertionFacet(test="$value > \'1999-12-31T23:59:59\'"):\n\nReason: \'>\' operator at line 1, column 8: [err:XPTY0004] \'>\' not supported between instances of \'DateTime\' and \'str\'\n\nSchema:\n\n  <xs:assertion xmlns:xs="http://www.w3.org/2001/XMLSchema" test="$value &gt; \'1999-12-31T23:59:59\'" />\n'

tests/validators/test_simple_types.py:232: AssertionError
=========================== short test summary info ============================
FAILED tests/validators/test_simple_types.py::TestXsd11SimpleTypes::test_assertion_facet
```

**Comment by fabaff** [comment]
Tests are working fine in a clean `venv` on Fedora 35 with Python 3.10.0.

```bash
======================================================================================================================================================== test session starts ========================================================================================================================================================
platform linux -- Python 3.10.0, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/fab/Documents/repos/xmlschema
collected 1186 items                                                                                                                                                                                                                                                                                                                

tests/test_cli.py .....................                                                                                                                                                                                                                                                                                       [  1%]
[...]
tests/validators/test_xsdbase.py ...........................................                                                                                                                                                                                                                                                  [100%]

================================================================================================================================================ 1070 passed, 116 skipped in 32.60s =================================================================================================================================================
```

**Comment by brunato** [comment]
Hi,
depends by the _elementpath_ version, which is the version of _elementpath_ in the failed case?

Thanks

**Comment by brunato** [comment]
I see, you have packed with a previous version of *xmlschema*, the test is already fixed in latest release (v1.8.2), skipping check the error message, that has changed with latest releases of _elementpath_.

**Comment by fabaff** [comment]
Thanks @brunato, you are right. The issue was a combination of different versions. With `elementpath-2.4.0` and `xmlschema-1.8.2` the builds are passing on NixOS as well.

Thanks for your support and enjoy your weekend.
